HopsanGUI/LogDataHandler2.cpp

```diff
@@ -996,6 +995,101 @@ void LogDataHandler2::importFromPlainColumnCsv(QString importFilePath, const QCh
     limitPlotGenerations();
 }
 
+void LogDataHandler2::importFromPlainRowCsv(QString importFilePath, const QChar separator, const int linesToSkip, const int timeRow)
+{
+    if(importFilePath.isEmpty())
+    {
+        importFilePath = QFileDialog::getOpenFileName(0,tr("Choose .csv File"),
+                                                       gpConfig->getStringSetting(CFG_PLOTDATADIR),
+                                                       tr("Comma-separated values files (*.csv)"));
+    }
+    if(importFilePath.isEmpty())
+    {
+        return;
+    }
+
+    QFile file(importFilePath);
+    QFileInfo fileInfo(file);
+    gpConfig->setStringSetting(CFG_PLOTDATADIR, fileInfo.absolutePath());
+
+    bool parseOk = true;
+
+    if (file.open(QFile::ReadOnly))
+    {
+        QStringList allNames, allAlias, allUnits;
+        QList< QVector<double> > allDatas;
+        int nDataValueElements = -1;
+        QTextStream contentStream(&file);
+        while (!contentStream.atEnd())
+        {
+            //! @todo this is not the most clever and speedy implementation
+            QStringList lineFields = contentStream.readLine().split(separator);
+            if (lineFields.size() > linesToSkip)
+            {
+                // Read metadata
+                if(linesToSkip > 0) {
+                    allNames.append(lineFields[0]);
+                }
+
+                allDatas.append(QVector<double>());
+                QVector<double> &data = allDatas.last();
+                if (nDataValueElements > -1)
+                {
+                    data.reserve(nDataValueElements);
+                }
+                for (int i=linesToSkip; i<lineFields.size(); ++i)
+                {
+                    bool ok;
+                    data.append(lineFields[i].toDouble(&ok));
+                    if(!ok)
+                    {
+                        parseOk = false;
+                    }
+                }
+                nDataValueElements = data.size();
+            }
+        }
+
+        if (parseOk)
+        {
+            ++mCurrentGenerationNumber;
+
+            // Figure out time
+            SharedVectorVariableT pTimeVec;
+            //! @todo what if multiple subsystems with different time
+            pTimeVec = insertTimeVectorVariable(allDatas[timeRow], fileInfo.absoluteFilePath());
```

Comment:
What if there is no time row in the data, (timeRow = -1) might be used to indicate this. Then a time vector should not be created

Reply:
I noticed that this may also be a problem in the existing column reader code, not sure if it needs fixing right now, probably not

Reply:
I fixed it for the row-wise function only. We can add the other fix later if needed.